### PR TITLE
audio attachment viewer

### PR DESCRIPTION
**Pull request by mmaxim**
Play attachments with a custom player, and render better audio visualization.

### CODE REVIEW

File: go/chat/attachments/sender.go [hunk 1]
```diff
@@ -67,6 +67,25 @@ func (s *Sender) makeBaseAttachmentMessage(ctx context.Context, tlfName string,
 	} else {
 		outboxID = *inOutboxID
 	}
+
+	if ephemeralLifetime != nil {
```
Comment (joshblum) [inline go/chat/attachments/sender.go]: this shouldn't move, msg is redefined on L89 so this value would be blown away

File: go/chat/utils/utils.go [hunk 2]
```diff
@@ -1061,7 +1061,11 @@ func GetMsgSnippetBody(msg chat1.MessageUnboxed) (snippet string) {
 			case chat1.AssetMetadataType_IMAGE:
 				title = "📷 attachment"
 			case chat1.AssetMetadataType_VIDEO:
-				title = "🎞 attachment"
+				if obj.Metadata.Video().IsAudio {
+					title = "🔊 attachment"
```
Comment (joshblum) [inline go/chat/utils/utils.go]: 🎉 👏 , was going to suggest this

File: shared/constants/types/chat2/index.tsx [hunk 3]
```diff
@@ -130,9 +131,10 @@ export type AudioRecordingInfo = {
   status: AudioRecordingStatus
   outboxID: Buffer
   path: string
+  recordEnd?: number
   recordStart: number
   isLocked: boolean
-  amps: Array<number>
+  amps?: AmpTracker
```
Comment (songgao) [inline shared/constants/types/chat2/index.tsx]: `AmpTracker` is a class that mutates itself. Should probably turn it into an object and move respective operations into the reducer?

Comment (mmaxim) [reply shared/constants/types/chat2/index.tsx]: I don't want any of this stuff to go through Redux

**Comment by mmaxim** [comment]
I rolled the audio library update into this PR as well.

File: shared/constants/types/chat2/index.tsx (hunk 3 above)
Comment (songgao) [reply shared/constants/types/chat2/index.tsx]: Maybe just store it separately then? Kind of weird to put something not mutated by reducer inside the store.

**Pull request merged**